src/sql/parts/notebook/notebook.component.ts

```diff
@@ -313,6 +318,16 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		this._changeRef.detectChanges();
 	}
 
+	private handleProviderIdChanged(providerId: string) {
+		// If there are any actions that were related to the previous provider,
+		// disable them in the actionBar
+		this._providerRelatedActions.forEach(action => {
+			action.enabled = false;
+		});
+		this.setContextKeyServiceWithProviderId(providerId);
```

Comment:
Minor: maybe merge these two / create a wrapper call that is `updateProviderAndRefreshActions`